crates/net/network-types/src/peers/config.rs

```diff
@@ -2,9 +2,11 @@
 
 use crate::{BackoffKind, ReputationChangeWeights};
 use reth_net_banlist::BanList;
-use reth_network_peers::NodeRecord;
+use reth_network_peers::{NodeRecord, TrustedPeer};
+use serde::de::{self, Deserializer, SeqAccess, Visitor};
```

Comment:
this needs to be gated behind the serde feature flag

```suggestion
#[cfg(feature = "serde")]
use serde::de::{self, Deserializer, SeqAccess, Visitor};
```